packages/contracts/contracts/appDefinitions/HighRollerApp.sol

```diff
@@ -0,0 +1,141 @@
+pragma solidity 0.4.25;
+pragma experimental "ABIEncoderV2";
+
+import "../lib/Transfer.sol";
+
+
+contract HighRollerApp {
+
+  enum ActionType {
+    COMMIT_TO_HASH,
+    COMMIT_TO_NUM,
+    REVEAL
+  }
+
+  enum Stage {
+    COMMITING_HASH, COMMITTING_NUM, REVEALING, DONE
+  }
+
+  enum Player {
+    FIRST,
+    SECOND,
+    TIE // TODO How should we handle this? Doesn't make sense that it is a "Player"
+  }
+
+  struct AppState {
+    address[2] playerAddrs;
+    Stage stage;
+    bytes32 commitHash;
+    uint256 commitNum;
+    uint256[2] playerFirstRoll; // TODO Does it make sense to store these in AppState?
+    uint256[2] playerSecondRoll;
+    Player winner;
+  }
+
+  struct Action {
+    ActionType actionType;
+    uint256 number;
+    bytes32 hash;
```

Comment:
I thought `hash` was a reserved keyword

Reply:
Should this be changed here and in the `CommitRevealApp`?

Reply:
Yes